backend/query/src/app.controller.ts

```diff
@@ -0,0 +1,50 @@
+import { InjectModel } from '@m8a/nestjs-typegoose';
+import {
+  ClassSerializerInterceptor,
+  Controller,
+  Get,
+  Logger,
+  SerializeOptions,
+  UseInterceptors,
+} from '@nestjs/common';
+import { ApiOkResponse, ApiOperation, ApiTags } from '@nestjs/swagger';
+import { ReturnModelType } from '@typegoose/typegoose';
+import { EditSuggestion } from './models/edit-suggestion.model';
+import { Item } from './models/item.model';
+import { GlobalStatistics } from './shared/interfaces/global-statistics';
+
+@Controller()
+@ApiTags('app')
+@UseInterceptors(ClassSerializerInterceptor)
+@SerializeOptions({ strategy: 'excludeAll' })
+export class AppController {
+  private readonly logger = new Logger(AppController.name);
+
+  constructor(
+    @InjectModel(Item)
+    private readonly itemModel: ReturnModelType<typeof Item>,
+    @InjectModel(EditSuggestion)
+    private readonly editSuggestionModel: ReturnModelType<
+      typeof EditSuggestion
+    >,
+  ) {}
+
+  @Get('statistics')
+  @ApiOperation({ summary: 'Get global statistics' })
+  @ApiOkResponse({
+    type: GlobalStatistics,
+    description: `Global statistics, the totalContribution ignores hard deleted items 
+      We can argue that 1 item creation + 1 item deletion = 1 falsy contribution`,
+  })
+  // TODO: Debate if this should use the eventStoreConnection to get more accurate counts, for example if approved edit suggestions get deleted
+  async getGlobalStatistics() {
+    this.logger.log(`Get global statistics`);
+    // TODO: Add filter if items can be softdeleted
+    const totalItems = await this.itemModel.count().exec();
+    const totalEditContributions = await this.editSuggestionModel
```

Comment:
Ergänzend zu Frage vorhin: Ist das nicht besser, das aus den Profiles rauszulesen (wo es schon drinsteht)? Wieso sollten die suggestions ins query backend?

Reply:
die stehen nicht im profile drin...
Wenn es keinen use Case gibt die ausm profil zu holen warum sollten sie dann da drin als Duplicates gespeichert werden>

Reply:
@Branchverse Lass mal quatschen

Reply:
Ich sehe keinen aktuten Sinn darin die Items direkt in Profile zu packen. Als Liste gespeichert müssen die sowieso Iwo existieren und im Profil ids zu speichern macht keinen Sinn (ist ja kein RDBMS)

Sprich das so über die counts abzuhandeln scheint mir das sinnvollste. Abgesehen ist so die liste deutlich besser zählbar 🤔